service/src/main/kotlin/fi/espoo/evaka/shared/security/Action.kt

```diff
@@ -261,7 +261,7 @@ sealed interface Action {
         CREATE_NOTE(HasGlobalRole(SERVICE_WORKER), HasUnitRole(SPECIAL_EDUCATION_TEACHER).inPreferredUnitOfApplication(), HasUnitRole(SPECIAL_EDUCATION_TEACHER).inPlacementPlanUnitOfApplication()),
         READ_SPECIAL_EDUCATION_TEACHER_NOTES(HasUnitRole(SPECIAL_EDUCATION_TEACHER).inPreferredUnitOfApplication(), HasUnitRole(SPECIAL_EDUCATION_TEACHER).inPlacementPlanUnitOfApplication()),
 
-        READ_ATTACHMENTS(HasGlobalRole(SERVICE_WORKER)),
+        READ_ATTACHMENTS(HasGlobalRole(SERVICE_WORKER), HasUnitRole(UNIT_SUPERVISOR).inPlacementPlanUnitWithExtendedCare()),
```

Comment:
This rule gives access to *all application attachments* for supervisors in extended care units. Isn't the requirement to limit access to only attachments with type `EXTENDED_CARE`?

If so, I think a different approach is needed. My suggestion: remove the READ_ATTACHMENTS Action completely, and instead filter attachments using one `checkPermissionFor` batch operation in the controller.

So, instead of

```kotlin
val attachments =
    if (accessControl.hasPermissionFor(user, Action.Application.READ_ATTACHMENTS, applicationId)) {
        tx.getApplicationAttachments(applicationId)
    } else {
        listOf()
    }
```

do something like this (code is untested):

```kotlin
val attachments = tx.getApplicationAttachments(applicationId).let { allAttachments ->
  val decisions = accessControl.checkPermissionFor(user, Action.Attachment.READ_APPLICATION_ATTACHMENT, allAttachments.map { it.id })
  allAttachments.filter { decisions[it.id].isPermitted() }
}
```

READ_APPLICATION_ATTACHMENT already includes a rule that handles both the `round_the_clock` flag in the daycare *and* the `EXTENDED_CARE` type of the attachment. The only downside is that we fetch the attachments even if the user can't have access to any of them, but this seems like an acceptable trade-off to me.

Reply:
Ah right! That seems to be the proper action to look 👍  Maybe we could just filter `allAttachments` by checking that user has permission for `READ_APPLICATION_ATTACHMENT` for that attachment 🤔